diff --git a/menudata_mine.c b/menudata_mate.c
index 5aa72b0..75f5e46 100644
--- a/menudata_mine.c
+++ b/menudata_mate.c
@@ -6,9 +5,0 @@
-#if 0
-// ColorMode 是一个在OLED_UI当中定义的bool类型变量，用于控制OLED显示的颜色模式， DARKMODE 为深色模式， LIGHTMOOD 为浅色模式。这里将其引出是为了创建单选框菜单项。
-extern bool ColorMode;
-extern bool OLED_UI_ShowFps;
-// OLED_UI_Brightness 是一个在OLED_UI当中定义的int16_t类型变量，用于控制OLED显示的亮度。这里将其引出是为了创建调整亮度的滑动条窗口，范围0-255。
-extern int16_t OLED_UI_Brightness;
-float testfloatnum = 0.5;
-int32_t testintnum = 1;
-#endif
@@ -17 +8,2 @@ int32_t testintnum = 1;
-bool oled_ui_exit = false;
+bool oled_ui_exit_save = false;
+bool oled_ui_exit_cancel = false;
@@ -24,3 +16,50 @@ bool oled_ui_exit = false;
-#if 0
-//关于窗口的结构体
-MenuWindow SetBrightnessWindow = {
+
+
+
+//主LOGO移动的结构体
+OLED_ChangePoint LogoMove;
+//主LOGO文字移动的结构体
+OLED_ChangePoint LogoTextMove;
+//welcome文字移动的结构体
+OLED_ChangePoint WelcomeTextMove;
+
+extern OLED_ChangePoint OLED_UI_PageStartPoint ;
+
+
+//主菜单的辅助显示函数
+void MainAuxFunc(void){
+	//不显示
+	LogoMove.TargetPoint.X = -200;
+	LogoMove.TargetPoint.Y = 0;
+	LogoMove.CurrentPoint.X = -200;
+	LogoMove.CurrentPoint.Y = 0;
+
+	LogoTextMove.TargetPoint.X = 129;
+	LogoTextMove.TargetPoint.Y = 0;
+	LogoTextMove.CurrentPoint.X = 129;
+	LogoTextMove.CurrentPoint.Y = 0;
+	
+	WelcomeTextMove.TargetPoint.X = 128;
+	WelcomeTextMove.TargetPoint.Y = 0;
+	WelcomeTextMove.CurrentPoint.X = 128;
+	WelcomeTextMove.CurrentPoint.Y = 0;
+}
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+/*上面为示例，下面依照示例创建自己的窗口变量和函数 关于窗口的结构体*/
+MenuWindow R_ChuFa_x_Window = {
@@ -29 +68 @@ MenuWindow SetBrightnessWindow = {
-	.Text_String = "屏幕亮度屏幕亮度",					//窗口标题
+	.Text_String = "R_ChuFa_x",					//窗口标题
@@ -35,5 +74,4 @@ MenuWindow SetBrightnessWindow = {
-
-	.Prob_Data_Int_16 = &OLED_UI_Brightness,				//显示的变量地址
-	.Prob_DataStep = 5,								//步长
-	.Prob_MinData = 5,									//最小值
-	.Prob_MaxData = 255, 								//最大值
+	.Prob_Data_Int_32 = &STORAGE_Data.R_ChuFa_x,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = -150,									//最小值
+	.Prob_MaxData = 150, 								//最大值
@@ -42 +80 @@ MenuWindow SetBrightnessWindow = {
-	.Prob_SideDistance = 4,								//边距
+	.Prob_SideDistance = 4,	
@@ -44,5 +82,2 @@ MenuWindow SetBrightnessWindow = {
-/**
- * @brief 创建显示亮度窗口
- */
-void BrightnessWindow(void){
-	OLED_UI_CreateWindow(&SetBrightnessWindow);
+void Show_R_ChuFa_x_Window(void){
+	OLED_UI_CreateWindow(&R_ChuFa_x_Window);
@@ -50,2 +85 @@ void BrightnessWindow(void){
-//关于窗口的结构体
-MenuWindow NullWindow = {
+MenuWindow R_ChuFa_y_Window = {
@@ -53,0 +88,4 @@ MenuWindow NullWindow = {
+	.Text_String = "R_ChuFa_y",					//窗口标题
+	.Text_FontSize = OLED_UI_FONT_12,				//字高
+	.Text_FontSideDistance = 4,							//字体距离左侧的距离
+	.Text_FontTopDistance = 3,							//字体距离顶部的距离
@@ -55,0 +94,7 @@ MenuWindow NullWindow = {
+	.Prob_Data_Int_32 = &STORAGE_Data.R_ChuFa_y,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = 0,									//最小值
+	.Prob_MaxData = 550, 								//最大值
+	.Prob_BottomDistance = 3,							//底部间距
+	.Prob_LineHeight = 8,								//进度条高度
+	.Prob_SideDistance = 4,	
@@ -57,5 +102,2 @@ MenuWindow NullWindow = {
-/**
- * @brief 创建显示亮度窗口
- */
-void EmptyWindow(void){
-	OLED_UI_CreateWindow(&NullWindow);
+void Show_R_ChuFa_y_Window(void){
+	OLED_UI_CreateWindow(&R_ChuFa_y_Window);
@@ -63,4 +105,7 @@ void EmptyWindow(void){
-//关于窗口的结构体
-MenuWindow TextWindow = {
-	.General_Width = 75,								//窗口宽度
-	.General_Height = 18, 							//窗口高度
+MenuWindow B_ChuFa_x_Window = {
+	.General_Width = 80,								//窗口宽度
+	.General_Height = 28, 							//窗口高度
+	.Text_String = "B_ChuFa_x",					//窗口标题
+	.Text_FontSize = OLED_UI_FONT_12,				//字高
+	.Text_FontSideDistance = 4,							//字体距离左侧的距离
+	.Text_FontTopDistance = 3,							//字体距离顶部的距离
@@ -69,7 +114,7 @@ MenuWindow TextWindow = {
-
-	.Text_String = "文字text",					//窗口标题
-	.Text_FontSize = OLED_UI_FONT_12,				//字高
-	.Text_FontSideDistance = 8,							//字体距离左侧的距离
-	.Text_FontTopDistance = 2,							//字体距离顶部的距离
-	
-
+	.Prob_Data_Int_32 = &STORAGE_Data.B_ChuFa_x,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = -150,									//最小值
+	.Prob_MaxData = 150, 								//最大值
+	.Prob_BottomDistance = 3,							//底部间距
+	.Prob_LineHeight = 8,								//进度条高度
+	.Prob_SideDistance = 4,	
@@ -77,5 +122,2 @@ MenuWindow TextWindow = {
-/**
- * @brief 创建显示亮度窗口
- */
-void ShowTextWindow(void){
-	OLED_UI_CreateWindow(&TextWindow);
+void Show_B_ChuFa_x_Window(void){
+	OLED_UI_CreateWindow(&B_ChuFa_x_Window);
@@ -83,2 +125 @@ void ShowTextWindow(void){
-//关于窗口的结构体
-MenuWindow FloatDataWindow = {
+MenuWindow B_ChuFa_y_Window = {
@@ -87 +128 @@ MenuWindow FloatDataWindow = {
-	.Text_String = "浮点数据测试",					//窗口标题
+	.Text_String = "B_ChuFa_y",					//窗口标题
@@ -93,5 +134,4 @@ MenuWindow FloatDataWindow = {
-
-	.Prob_Data_Float = &testfloatnum,				//显示的变量地址
-	.Prob_DataStep = 0.1,								//步长
-	.Prob_MinData = -100,									//最小值
-	.Prob_MaxData = 100, 								//最大值
+	.Prob_Data_Int_32 = &STORAGE_Data.B_ChuFa_y,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = 0,									//最小值
+	.Prob_MaxData = 550, 								//最大值
@@ -102,5 +142,2 @@ MenuWindow FloatDataWindow = {
-/**
- * @brief 创建显示亮度窗口
- */
-void ShowFloatDataWindow(void){
-	OLED_UI_CreateWindow(&FloatDataWindow);
+void Show_B_ChuFa_y_Window(void){
+	OLED_UI_CreateWindow(&B_ChuFa_y_Window);
@@ -108,2 +145,4 @@ void ShowFloatDataWindow(void){
-//关于窗口的结构体
-MenuWindow IntDataWindow = {
+
+
+
+MenuWindow R_YuanPan_y_Window = {
@@ -112 +151 @@ MenuWindow IntDataWindow = {
-	.Text_String = "整数数据测试",					//窗口标题
+	.Text_String = "R_YuanPan_y",					//窗口标题
@@ -118,2 +157 @@ MenuWindow IntDataWindow = {
-
-	.Prob_Data_Int_32 = &testintnum,				//显示的变量地址
+	.Prob_Data_Int_32 = &STORAGE_Data.R_YuanPan_y,				//显示的变量地址
@@ -121,2 +159,2 @@ MenuWindow IntDataWindow = {
-	.Prob_MinData = -100,									//最小值
-	.Prob_MaxData = 100, 								//最大值
+	.Prob_MinData = -20,									//最小值
+	.Prob_MaxData = 20, 								//最大值
@@ -127,5 +165,2 @@ MenuWindow IntDataWindow = {
-/**
- * @brief 创建显示亮度窗口
- */
-void ShowIntDataWindow(void){
-	OLED_UI_CreateWindow(&IntDataWindow);
+void Show_R_YuanPan_y_Window(void){
+	OLED_UI_CreateWindow(&R_YuanPan_y_Window);
@@ -133,4 +168 @@ void ShowIntDataWindow(void){
-#endif
-/*上面为示例，下面依照示例创建自己的窗口变量和函数*/
-//关于窗口的结构体
-MenuWindow line_kp_Window = {
+MenuWindow B_YuanPan_y_Window = {
@@ -139 +171 @@ MenuWindow line_kp_Window = {
-	.Text_String = "line_kp",					//窗口标题
+	.Text_String = "B_YuanPan_y",					//窗口标题
@@ -145,5 +177,4 @@ MenuWindow line_kp_Window = {
-
-	.Prob_Data_Float = &STORAGE_Data.line_kp,				//显示的变量地址
-	.Prob_DataStep = 0.05,								//步长
-	.Prob_MinData = -5.0,									//最小值
-	.Prob_MaxData = 5.0, 								//最大值
+	.Prob_Data_Int_32 = &STORAGE_Data.B_YuanPan_y,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = -20,									//最小值
+	.Prob_MaxData = 20, 								//最大值
@@ -154,5 +185,2 @@ MenuWindow line_kp_Window = {
-/**
- * @brief 创建显示line_kp窗口
- */
-void Show_line_kp_Window(void){
-	OLED_UI_CreateWindow(&line_kp_Window);
+void Show_B_YuanPan_y_Window(void){
+	OLED_UI_CreateWindow(&B_YuanPan_y_Window);
@@ -161 +189,4 @@ void Show_line_kp_Window(void){
-MenuWindow line_ki_Window = {
+
+
+
+MenuWindow C_GetTarget_y_Window = {
@@ -164 +195 @@ MenuWindow line_ki_Window = {
-	.Text_String = "line_ki",					//窗口标题
+	.Text_String = "C_GetTarget_y",					//窗口标题
@@ -170,5 +201,4 @@ MenuWindow line_ki_Window = {
-
-	.Prob_Data_Float = &STORAGE_Data.line_ki,				//显示的变量地址
-	.Prob_DataStep = 0.05,								//步长
-	.Prob_MinData = -5.0,									//最小值
-	.Prob_MaxData = 5.0, 								//最大值
+	.Prob_Data_Int_32 = &STORAGE_Data.C_GetTarget_y,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = 0,									//最小值
+	.Prob_MaxData = 200, 								//最大值
@@ -179,5 +209,2 @@ MenuWindow line_ki_Window = {
-/**
- * @brief 创建显示line_ki窗口
- */
-void Show_line_ki_Window(void){
-	OLED_UI_CreateWindow(&line_ki_Window);
+void Show_C_GetTarget_y_Window(void){
+	OLED_UI_CreateWindow(&C_GetTarget_y_Window);
@@ -185,2 +212 @@ void Show_line_ki_Window(void){
-
-MenuWindow line_kd_Window = {
+MenuWindow C_GetTarget_x_Window = {
@@ -189 +215 @@ MenuWindow line_kd_Window = {
-	.Text_String = "line_kd",					//窗口标题
+	.Text_String = "C_GetTarget_x",					//窗口标题
@@ -195,5 +221,4 @@ MenuWindow line_kd_Window = {
-
-	.Prob_Data_Float = &STORAGE_Data.line_kd,				//显示的变量地址
-	.Prob_DataStep = 0.01,								//步长
-	.Prob_MinData = -5.0,									//最小值
-	.Prob_MaxData = 5.0, 								//最大值
+	.Prob_Data_Int_32 = &STORAGE_Data.C_GetTarget_x,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = -100,									//最小值
+	.Prob_MaxData = 0, 								//最大值
@@ -204,5 +229,2 @@ MenuWindow line_kd_Window = {
-/**
- * @brief 创建显示line_kd窗口
- */
-void Show_line_kd_Window(void){
-	OLED_UI_CreateWindow(&line_kd_Window);
+void Show_C_GetTarget_x_Window(void){
+	OLED_UI_CreateWindow(&C_GetTarget_x_Window);
@@ -211 +233,5 @@ void Show_line_kd_Window(void){
-MenuWindow angle_kp_Window = {
+
+
+
+/*阶梯相关*/
+MenuWindow C_JieTi_Left_x_Window = {
@@ -214 +240 @@ MenuWindow angle_kp_Window = {
-	.Text_String = "angle_kp",					//窗口标题
+	.Text_String = "C_JieTi_Left_x",				//窗口标题
@@ -220,5 +246,4 @@ MenuWindow angle_kp_Window = {
-
-	.Prob_Data_Float = &STORAGE_Data.angle_kp,				//显示的变量地址
-	.Prob_DataStep = 0.05,								//步长
-	.Prob_MinData = -5.0,									//最小值
-	.Prob_MaxData = 5.0, 								//最大值
+	.Prob_Data_Int_32 = &STORAGE_Data.C_JieTi_Left_x,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = -60,									//最小值
+	.Prob_MaxData = 60, 								//最大值
@@ -227 +252 @@ MenuWindow angle_kp_Window = {
-	.Prob_SideDistance = 4,	
+	.Prob_SideDistance = 4,
@@ -229,5 +254,2 @@ MenuWindow angle_kp_Window = {
-/**
- * @brief 创建显示angle_kp窗口
- */
-void Show_angle_kp_Window(void){
-	OLED_UI_CreateWindow(&angle_kp_Window);
+void Show_C_JieTi_Left_x_Window(void){
+	OLED_UI_CreateWindow(&C_JieTi_Left_x_Window);
@@ -235,2 +257 @@ void Show_angle_kp_Window(void){
-
-MenuWindow angle_ki_Window = {
+MenuWindow C_Jieti_FwdTarget_y_Window = {
@@ -239 +260 @@ MenuWindow angle_ki_Window = {
-	.Text_String = "angle_ki",					//窗口标题
+	.Text_String = "C_Jieti_FwdTarget_y",		//窗口标题
@@ -245,5 +266,4 @@ MenuWindow angle_ki_Window = {
-
-	.Prob_Data_Float = &STORAGE_Data.angle_ki,				//显示的变量地址
-	.Prob_DataStep = 0.05,								//步长
-	.Prob_MinData = -5.0,									//最小值
-	.Prob_MaxData = 5.0, 								//最大值
+	.Prob_Data_Int_32 = &STORAGE_Data.C_Jieti_FwdTarget_y,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = 0,									//最小值
+	.Prob_MaxData = 200, 								//最大值
@@ -252 +272 @@ MenuWindow angle_ki_Window = {
-	.Prob_SideDistance = 4,	
+	.Prob_SideDistance = 4,
@@ -254,5 +274,2 @@ MenuWindow angle_ki_Window = {
-/**
- * @brief 创建显示angle_ki窗口
- */
-void Show_angle_ki_Window(void){
-	OLED_UI_CreateWindow(&angle_ki_Window);
+void Show_C_Jieti_FwdTarget_y_Window(void){
+	OLED_UI_CreateWindow(&C_Jieti_FwdTarget_y_Window);
@@ -260,2 +277 @@ void Show_angle_ki_Window(void){
-
-MenuWindow angle_kd_Window = {
+MenuWindow C_JitTi_Step_x_Window = {
@@ -264 +280 @@ MenuWindow angle_kd_Window = {
-	.Text_String = "angle_kd",					//窗口标题
+	.Text_String = "C_JitTi_Step_x",				//窗口标题
@@ -270,5 +286,4 @@ MenuWindow angle_kd_Window = {
-
-	.Prob_Data_Float = &STORAGE_Data.angle_kd,				//显示的变量地址
-	.Prob_DataStep = 0.01,								//步长
-	.Prob_MinData = -5.0,									//最小值
-	.Prob_MaxData = 5.0, 								//最大值
+	.Prob_Data_Int_32 = &STORAGE_Data.C_JitTi_Step_x,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = -60,									//最小值
+	.Prob_MaxData = 60, 								//最大值
@@ -277 +292 @@ MenuWindow angle_kd_Window = {
-	.Prob_SideDistance = 4,	
+	.Prob_SideDistance = 4,
@@ -279,5 +294,2 @@ MenuWindow angle_kd_Window = {
-/**
- * @brief 创建显示angle_kd窗口
- */
-void Show_angle_kd_Window(void){
-	OLED_UI_CreateWindow(&angle_kd_Window);
+void Show_C_JitTi_Step_x_Window(void){
+	OLED_UI_CreateWindow(&C_JitTi_Step_x_Window);
@@ -285,2 +297 @@ void Show_angle_kd_Window(void){
-
-MenuWindow speed_left_kp_Window = {
+MenuWindow C_Jieti_Cross_x_Window = {
@@ -289 +300 @@ MenuWindow speed_left_kp_Window = {
-	.Text_String = "speed_left_kp",					//窗口标题
+	.Text_String = "C_Jieti_Cross_x",				//窗口标题
@@ -295,5 +306,4 @@ MenuWindow speed_left_kp_Window = {
-
-	.Prob_Data_Float = &STORAGE_Data.speed_left_kp,				//显示的变量地址
-	.Prob_DataStep = 0.05,								//步长
-	.Prob_MinData = -5.0,									//最小值
-	.Prob_MaxData = 5.0, 								//最大值
+	.Prob_Data_Int_32 = &STORAGE_Data.C_Jieti_Cross_x,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = -60,									//最小值
+	.Prob_MaxData = 60, 								//最大值
@@ -302 +312 @@ MenuWindow speed_left_kp_Window = {
-	.Prob_SideDistance = 4,	
+	.Prob_SideDistance = 4,
@@ -304,5 +314,2 @@ MenuWindow speed_left_kp_Window = {
-/**
- * @brief 创建显示speed_left_kp窗口
- */
-void Show_speed_left_kp_Window(void){
-	OLED_UI_CreateWindow(&speed_left_kp_Window);
+void Show_C_Jieti_Cross_x_Window(void){
+	OLED_UI_CreateWindow(&C_Jieti_Cross_x_Window);
@@ -310,2 +317 @@ void Show_speed_left_kp_Window(void){
-
-MenuWindow speed_left_ki_Window = {
+MenuWindow C_JieTi_JiaoZhun_x1_Window = {
@@ -314 +320 @@ MenuWindow speed_left_ki_Window = {
-	.Text_String = "speed_left_ki",					//窗口标题
+	.Text_String = "C_JieTi_JiaoZhun_x1",		//窗口标题
@@ -320,5 +326,4 @@ MenuWindow speed_left_ki_Window = {
-
-	.Prob_Data_Float = &STORAGE_Data.speed_left_ki,				//显示的变量地址
-	.Prob_DataStep = 0.01,								//步长
-	.Prob_MinData = -5.0,									//最小值
-	.Prob_MaxData = 5.0, 								//最大值
+	.Prob_Data_Int_32 = &STORAGE_Data.C_JieTi_JiaoZhun_x1,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = -30,									//最小值
+	.Prob_MaxData = 30, 								//最大值
@@ -327 +332 @@ MenuWindow speed_left_ki_Window = {
-	.Prob_SideDistance = 4,	
+	.Prob_SideDistance = 4,
@@ -329,5 +334,2 @@ MenuWindow speed_left_ki_Window = {
-/**
- * @brief 创建显示speed_left_ki窗口
- */
-void Show_speed_left_ki_Window(void){
-	OLED_UI_CreateWindow(&speed_left_ki_Window);
+void Show_C_JieTi_JiaoZhun_x1_Window(void){
+	OLED_UI_CreateWindow(&C_JieTi_JiaoZhun_x1_Window);
@@ -335,2 +337 @@ void Show_speed_left_ki_Window(void){
-
-MenuWindow speed_left_kd_Window = {
+MenuWindow C_JieTi_JiaoZhun_x2_Window = {
@@ -339 +340 @@ MenuWindow speed_left_kd_Window = {
-	.Text_String = "speed_left_kd",					//窗口标题
+	.Text_String = "C_JieTi_JiaoZhun_x2",		//窗口标题
@@ -345,5 +346,4 @@ MenuWindow speed_left_kd_Window = {
-
-	.Prob_Data_Float = &STORAGE_Data.speed_left_kd,				//显示的变量地址
-	.Prob_DataStep = 0.05,								//步长
-	.Prob_MinData = -5.0,									//最小值
-	.Prob_MaxData = 5.0, 								//最大值
+	.Prob_Data_Int_32 = &STORAGE_Data.C_JieTi_JiaoZhun_x2,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = -30,									//最小值
+	.Prob_MaxData = 30, 								//最大值
@@ -352 +352 @@ MenuWindow speed_left_kd_Window = {
-	.Prob_SideDistance = 4,	
+	.Prob_SideDistance = 4,
@@ -354,5 +354,2 @@ MenuWindow speed_left_kd_Window = {
-/**
- * @brief 创建显示speed_left_kd窗口
- */
-void Show_speed_left_kd_Window(void){
-	OLED_UI_CreateWindow(&speed_left_kd_Window);
+void Show_C_JieTi_JiaoZhun_x2_Window(void){
+	OLED_UI_CreateWindow(&C_JieTi_JiaoZhun_x2_Window);
@@ -360,2 +357 @@ void Show_speed_left_kd_Window(void){
-
-MenuWindow speed_right_kp_Window = {
+MenuWindow C_JieTi_JiaoZhun_y1_Window = {
@@ -364 +360 @@ MenuWindow speed_right_kp_Window = {
-	.Text_String = "speed_right_kp",					//窗口标题
+	.Text_String = "C_JieTi_JiaoZhun_y1",		//窗口标题
@@ -370,5 +366,4 @@ MenuWindow speed_right_kp_Window = {
-
-	.Prob_Data_Float = &STORAGE_Data.speed_right_kp,				//显示的变量地址
-	.Prob_DataStep = 0.05,								//步长
-	.Prob_MinData = -5.0,									//最小值
-	.Prob_MaxData = 5.0, 								//最大值
+	.Prob_Data_Int_32 = &STORAGE_Data.C_JieTi_JiaoZhun_y1,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = -30,									//最小值
+	.Prob_MaxData = 30, 								//最大值
@@ -377 +372 @@ MenuWindow speed_right_kp_Window = {
-	.Prob_SideDistance = 4,	
+	.Prob_SideDistance = 4,
@@ -379,5 +374,2 @@ MenuWindow speed_right_kp_Window = {
-/**
- * @brief 创建显示speed_right_kp窗口
- */
-void Show_speed_right_kp_Window(void){
-	OLED_UI_CreateWindow(&speed_right_kp_Window);
+void Show_C_JieTi_JiaoZhun_y1_Window(void){
+	OLED_UI_CreateWindow(&C_JieTi_JiaoZhun_y1_Window);
@@ -385,2 +377 @@ void Show_speed_right_kp_Window(void){
-
-MenuWindow speed_right_ki_Window = {
+MenuWindow C_JieTi_JiaoZhun_y2_Window = {
@@ -389 +380 @@ MenuWindow speed_right_ki_Window = {
-	.Text_String = "speed_right_ki",					//窗口标题
+	.Text_String = "C_JieTi_JiaoZhun_y2",		//窗口标题
@@ -395,5 +386,4 @@ MenuWindow speed_right_ki_Window = {
-
-	.Prob_Data_Float = &STORAGE_Data.speed_right_ki,				//显示的变量地址
-	.Prob_DataStep = 0.01,								//步长
-	.Prob_MinData = -5.0,									//最小值
-	.Prob_MaxData = 5.0, 								//最大值
+	.Prob_Data_Int_32 = &STORAGE_Data.C_JieTi_JiaoZhun_y2,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = -30,									//最小值
+	.Prob_MaxData = 30, 								//最大值
@@ -402 +392 @@ MenuWindow speed_right_ki_Window = {
-	.Prob_SideDistance = 4,	
+	.Prob_SideDistance = 4,
@@ -404,5 +394,2 @@ MenuWindow speed_right_ki_Window = {
-/**
- * @brief 创建显示speed_right_ki窗口
- */
-void Show_speed_right_ki_Window(void){
-	OLED_UI_CreateWindow(&speed_right_ki_Window);
+void Show_C_JieTi_JiaoZhun_y2_Window(void){
+	OLED_UI_CreateWindow(&C_JieTi_JiaoZhun_y2_Window);
@@ -411 +398,5 @@ void Show_speed_right_ki_Window(void){
-MenuWindow speed_right_kd_Window = {
+
+
+
+/*立桩相关*/
+MenuWindow R_LiZhuang_Delay_x_Window = {
@@ -414 +405 @@ MenuWindow speed_right_kd_Window = {
-	.Text_String = "speed_right_kd",					//窗口标题
+	.Text_String = "R_LiZhuang_Delay_x",		//窗口标题
@@ -420,5 +411,4 @@ MenuWindow speed_right_kd_Window = {
-
-	.Prob_Data_Float = &STORAGE_Data.speed_right_kd,				//显示的变量地址
-	.Prob_DataStep = 0.05,								//步长
-	.Prob_MinData = -5.0,									//最小值
-	.Prob_MaxData = 5.0, 								//最大值
+	.Prob_Data_Int_32 = &STORAGE_Data.R_LiZhuang_Delay_x,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = 0,									//最小值
+	.Prob_MaxData = 1000, 								//最大值
@@ -427 +417 @@ MenuWindow speed_right_kd_Window = {
-	.Prob_SideDistance = 4,	
+	.Prob_SideDistance = 4,
@@ -429,5 +419,2 @@ MenuWindow speed_right_kd_Window = {
-/**
- * @brief 创建显示speed_right_kd窗口
- */
-void Show_speed_right_kd_Window(void){
-	OLED_UI_CreateWindow(&speed_right_kd_Window);
+void Show_R_LiZhuang_Delay_x_Window(void){
+	OLED_UI_CreateWindow(&R_LiZhuang_Delay_x_Window);
@@ -435,2 +422 @@ void Show_speed_right_kd_Window(void){
-
-MenuWindow angle_offset_Window = {
+MenuWindow B_LiZhuang_Delay_x_Window = {
@@ -439 +425 @@ MenuWindow angle_offset_Window = {
-	.Text_String = "angle_offset",					//窗口标题
+	.Text_String = "B_LiZhuang_Delay_x",		//窗口标题
@@ -445,5 +431,4 @@ MenuWindow angle_offset_Window = {
-
-	.Prob_Data_Float = &STORAGE_Data.angle_offset,				//显示的变量地址
-	.Prob_DataStep = 0.02,								//步长
-	.Prob_MinData = -5.0,									//最小值
-	.Prob_MaxData = 5.0, 								//最大值
+	.Prob_Data_Int_32 = &STORAGE_Data.B_LiZhuang_Delay_x,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = 0,									//最小值
+	.Prob_MaxData = 1000, 								//最大值
@@ -452 +437 @@ MenuWindow angle_offset_Window = {
-	.Prob_SideDistance = 4,	
+	.Prob_SideDistance = 4,
@@ -454,5 +439,2 @@ MenuWindow angle_offset_Window = {
-/**
- * @brief 创建显示angle_offset窗口
- */
-void Show_angle_offset_Window(void){
-	OLED_UI_CreateWindow(&angle_offset_Window);
+void Show_B_LiZhuang_Delay_x_Window(void){
+	OLED_UI_CreateWindow(&B_LiZhuang_Delay_x_Window);
@@ -460,2 +442 @@ void Show_angle_offset_Window(void){
-
-MenuWindow encoder_cnt_odd_Window = {
+MenuWindow C_LiZhuang_Target_y_Window = {
@@ -464 +445 @@ MenuWindow encoder_cnt_odd_Window = {
-	.Text_String = "encoder_cnt_odd",					//窗口标题
+	.Text_String = "C_LiZhuang_Target_y",		//窗口标题
@@ -470,5 +451,4 @@ MenuWindow encoder_cnt_odd_Window = {
-
-	.Prob_Data_Int_32 = &STORAGE_Data.encoder_cnt_odd,				//显示的变量地址
-	.Prob_DataStep = 5,								//步长
-	.Prob_MinData = 1500,									//最小值
-	.Prob_MaxData = 4500, 								//最大值
+	.Prob_Data_Int_32 = &STORAGE_Data.C_LiZhuang_Target_y,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = 0,									//最小值
+	.Prob_MaxData = 300, 								//最大值
@@ -477 +457 @@ MenuWindow encoder_cnt_odd_Window = {
-	.Prob_SideDistance = 4,	
+	.Prob_SideDistance = 4,
@@ -479,5 +459,2 @@ MenuWindow encoder_cnt_odd_Window = {
-/**
- * @brief 创建显示encoder_cnt_odd窗口
- */
-void Show_encoder_cnt_odd_Window(void){
-	OLED_UI_CreateWindow(&encoder_cnt_odd_Window);
+void Show_C_LiZhuang_Target_y_Window(void){
+	OLED_UI_CreateWindow(&C_LiZhuang_Target_y_Window);
@@ -485,2 +462 @@ void Show_encoder_cnt_odd_Window(void){
-
-MenuWindow encoder_cnt_even_Window = {
+MenuWindow C_LiZhuang_Fwd_y_Window = {
@@ -489 +465 @@ MenuWindow encoder_cnt_even_Window = {
-	.Text_String = "encoder_cnt_even",					//窗口标题
+	.Text_String = "C_LiZhuang_Fwd_y",			//窗口标题
@@ -495,5 +471,4 @@ MenuWindow encoder_cnt_even_Window = {
-
-	.Prob_Data_Int_32 = &STORAGE_Data.encoder_cnt_even,				//显示的变量地址
-	.Prob_DataStep = 5,								//步长
-	.Prob_MinData = 1500,									//最小值
-	.Prob_MaxData = 4500, 								//最大值
+	.Prob_Data_Int_32 = &STORAGE_Data.C_LiZhuang_Fwd_y,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = -60,									//最小值
+	.Prob_MaxData = 60, 								//最大值
@@ -502 +477 @@ MenuWindow encoder_cnt_even_Window = {
-	.Prob_SideDistance = 4,	
+	.Prob_SideDistance = 4,
@@ -504,5 +479,42 @@ MenuWindow encoder_cnt_even_Window = {
-/**
- * @brief 创建显示encoder_cnt_even窗口
- */
-void Show_encoder_cnt_even_Window(void){
-	OLED_UI_CreateWindow(&encoder_cnt_even_Window);
+void Show_C_LiZhuang_Fwd_y_Window(void){
+	OLED_UI_CreateWindow(&C_LiZhuang_Fwd_y_Window);
+}
+MenuWindow R_LiZhuang_Back_y_Window = {
+	.General_Width = 80,								//窗口宽度
+	.General_Height = 28, 							//窗口高度
+	.Text_String = "R_LiZhuang_Back_y",			//窗口标题
+	.Text_FontSize = OLED_UI_FONT_12,				//字高
+	.Text_FontSideDistance = 4,							//字体距离左侧的距离
+	.Text_FontTopDistance = 3,							//字体距离顶部的距离
+	.General_WindowType = WINDOW_ROUNDRECTANGLE, 	//窗口类型
+	.General_ContinueTime = 4.0,						//窗口持续时间
+	.Prob_Data_Int_32 = &STORAGE_Data.R_LiZhuang_Back_y,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = -60,									//最小值
+	.Prob_MaxData = 60, 								//最大值
+	.Prob_BottomDistance = 3,							//底部间距
+	.Prob_LineHeight = 8,								//进度条高度
+	.Prob_SideDistance = 4,
+};
+void Show_R_LiZhuang_Back_y_Window(void){
+	OLED_UI_CreateWindow(&R_LiZhuang_Back_y_Window);
+}
+MenuWindow B_LiZhuang_Back_y_Window = {
+	.General_Width = 80,								//窗口宽度
+	.General_Height = 28, 							//窗口高度
+	.Text_String = "B_LiZhuang_Back_y",			//窗口标题
+	.Text_FontSize = OLED_UI_FONT_12,				//字高
+	.Text_FontSideDistance = 4,							//字体距离左侧的距离
+	.Text_FontTopDistance = 3,							//字体距离顶部的距离
+	.General_WindowType = WINDOW_ROUNDRECTANGLE, 	//窗口类型
+	.General_ContinueTime = 4.0,						//窗口持续时间
+	.Prob_Data_Int_32 = &STORAGE_Data.B_LiZhuang_Back_y,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = -60,									//最小值
+	.Prob_MaxData = 60, 								//最大值
+	.Prob_BottomDistance = 3,							//底部间距
+	.Prob_LineHeight = 8,								//进度条高度
+	.Prob_SideDistance = 4,
+};
+void Show_B_LiZhuang_Back_y_Window(void){
+	OLED_UI_CreateWindow(&B_LiZhuang_Back_y_Window);
@@ -514,40 +526,20 @@ void Show_encoder_cnt_even_Window(void){
-
-
-
-
-
-
-
-
-
-
-//主LOGO移动的结构体
-OLED_ChangePoint LogoMove;
-//主LOGO文字移动的结构体
-OLED_ChangePoint LogoTextMove;
-//welcome文字移动的结构体
-OLED_ChangePoint WelcomeTextMove;
-
-extern OLED_ChangePoint OLED_UI_PageStartPoint ;
-
-
-
-//设置菜单项的辅助显示函数
-void SettingAuxFunc(void){
-	//在规定位置显示LOGO
-	if(fabs(OLED_UI_PageStartPoint.CurrentPoint.X - OLED_UI_PageStartPoint.TargetPoint.X) < 4){
-		LogoMove.TargetPoint.X = 0;
-		LogoMove.TargetPoint.Y = 0;
-	}
-	//将LOGOTEXT移动到屏幕右侧看不见的地方
-	LogoTextMove.TargetPoint.X = 129;
-	LogoTextMove.TargetPoint.Y = 0;
-	//将Welcome文字移动到屏幕底部看不见的地方
-	WelcomeTextMove.TargetPoint.X = 128;
-	WelcomeTextMove.TargetPoint.Y = 0;
-	ChangePoint(&LogoMove);
-	OLED_ShowImageArea(LogoMove.CurrentPoint.X,LogoMove.CurrentPoint.Y,32,64,0,0,128,128,OLED_UI_SettingsLogo);
-	ChangePoint(&LogoTextMove);
-	OLED_ShowImageArea(LogoTextMove.CurrentPoint.X,LogoTextMove.CurrentPoint.Y,26,64,0,0,128,128,OLED_UI_LOGOTEXT64);
-	ChangePoint(&WelcomeTextMove);
-	OLED_ShowImageArea(WelcomeTextMove.CurrentPoint.X,WelcomeTextMove.CurrentPoint.Y,16,64,0,0,128,128,OLED_UI_LOGOGithub);
+/*回家相关*/
+MenuWindow R_HuiJia_x_Window = {
+	.General_Width = 80,								//窗口宽度
+	.General_Height = 28, 							//窗口高度
+	.Text_String = "R_HuiJia_x",					//窗口标题
+	.Text_FontSize = OLED_UI_FONT_12,				//字高
+	.Text_FontSideDistance = 4,							//字体距离左侧的距离
+	.Text_FontTopDistance = 3,							//字体距离顶部的距离
+	.General_WindowType = WINDOW_ROUNDRECTANGLE, 	//窗口类型
+	.General_ContinueTime = 4.0,						//窗口持续时间
+	.Prob_Data_Int_32 = &STORAGE_Data.R_HuiJia_x,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = -100,									//最小值
+	.Prob_MaxData = 100, 								//最大值
+	.Prob_BottomDistance = 3,							//底部间距
+	.Prob_LineHeight = 8,								//进度条高度
+	.Prob_SideDistance = 4,
+};
+void Show_R_HuiJia_x_Window(void){
+	OLED_UI_CreateWindow(&R_HuiJia_x_Window);
@@ -555,15 +547,19 @@ void SettingAuxFunc(void){
-
-//关于菜单的辅助显示函数
-void AboutThisDeviceAuxFunc(void){
-	//将LOGO移动到屏幕上方看不见的地方
-	LogoMove.TargetPoint.X = 0;
-	LogoMove.TargetPoint.Y = -80;
-	ChangePoint(&LogoMove);
-    OLED_ShowImageArea(LogoMove.CurrentPoint.X,LogoMove.CurrentPoint.Y,32,64,0,0,128,128,OLED_UI_SettingsLogo);
-	//在屏幕右侧显示LOGO文字
-	if(fabs(OLED_UI_PageStartPoint.CurrentPoint.X - OLED_UI_PageStartPoint.TargetPoint.X) < 4){
-		LogoTextMove.TargetPoint.X = 102;
-		LogoTextMove.TargetPoint.Y = 0;
-	}
-	ChangePoint(&LogoTextMove);
-	OLED_ShowImageArea(LogoTextMove.CurrentPoint.X,LogoTextMove.CurrentPoint.Y,26,64,0,0,128,128,OLED_UI_LOGOTEXT64);
+MenuWindow B_HuiJia_x_Window = {
+	.General_Width = 80,								//窗口宽度
+	.General_Height = 28, 							//窗口高度
+	.Text_String = "B_HuiJia_x",					//窗口标题
+	.Text_FontSize = OLED_UI_FONT_12,				//字高
+	.Text_FontSideDistance = 4,							//字体距离左侧的距离
+	.Text_FontTopDistance = 3,							//字体距离顶部的距离
+	.General_WindowType = WINDOW_ROUNDRECTANGLE, 	//窗口类型
+	.General_ContinueTime = 4.0,						//窗口持续时间
+	.Prob_Data_Int_32 = &STORAGE_Data.B_HuiJia_x,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = -100,									//最小值
+	.Prob_MaxData = 100, 								//最大值
+	.Prob_BottomDistance = 3,							//底部间距
+	.Prob_LineHeight = 8,								//进度条高度
+	.Prob_SideDistance = 4,
+};
+void Show_B_HuiJia_x_Window(void){
+	OLED_UI_CreateWindow(&B_HuiJia_x_Window);
@@ -571,15 +567,19 @@ void AboutThisDeviceAuxFunc(void){
-//关于OLED UI的辅助显示函数
-void AboutOLED_UIAuxFunc(void){
-	//将LOGO移动到屏幕上方看不见的地方
-	LogoMove.TargetPoint.X = 0;
-	LogoMove.TargetPoint.Y = -80;
-	ChangePoint(&LogoMove);
-	OLED_ShowImageArea(LogoMove.CurrentPoint.X,LogoMove.CurrentPoint.Y,32,64,0,0,128,128,OLED_UI_SettingsLogo);
-	//在屏幕右测显示Welcome文字
-	if(fabs(OLED_UI_PageStartPoint.CurrentPoint.X - OLED_UI_PageStartPoint.TargetPoint.X) < 4){
-		WelcomeTextMove.TargetPoint.X = 110;
-		WelcomeTextMove.TargetPoint.Y = 0;
-	}
-	ChangePoint(&WelcomeTextMove);
-	OLED_ShowImageArea(WelcomeTextMove.CurrentPoint.X,WelcomeTextMove.CurrentPoint.Y,16,64,0,0,128,128,OLED_UI_LOGOGithub);
-
+MenuWindow C_HuiJia_y_Window = {
+	.General_Width = 80,								//窗口宽度
+	.General_Height = 28, 							//窗口高度
+	.Text_String = "C_HuiJia_y",					//窗口标题
+	.Text_FontSize = OLED_UI_FONT_12,				//字高
+	.Text_FontSideDistance = 4,							//字体距离左侧的距离
+	.Text_FontTopDistance = 3,							//字体距离顶部的距离
+	.General_WindowType = WINDOW_ROUNDRECTANGLE, 	//窗口类型
+	.General_ContinueTime = 4.0,						//窗口持续时间
+	.Prob_Data_Int_32 = &STORAGE_Data.C_HuiJia_y,				//显示的变量地址
+	.Prob_DataStep = 1,								//步长
+	.Prob_MinData = -60,									//最小值
+	.Prob_MaxData = 60, 								//最大值
+	.Prob_BottomDistance = 3,							//底部间距
+	.Prob_LineHeight = 8,								//进度条高度
+	.Prob_SideDistance = 4,
+};
+void Show_C_HuiJia_y_Window(void){
+	OLED_UI_CreateWindow(&C_HuiJia_y_Window);
@@ -587,7 +586,0 @@ void AboutOLED_UIAuxFunc(void){
-//主菜单的辅助显示函数
-void MainAuxFunc(void){
-	//不显示
-	LogoMove.TargetPoint.X = -200;
-	LogoMove.TargetPoint.Y = 0;
-	LogoMove.CurrentPoint.X = -200;
-	LogoMove.CurrentPoint.Y = 0;
@@ -595,10 +587,0 @@ void MainAuxFunc(void){
-	LogoTextMove.TargetPoint.X = 129;
-	LogoTextMove.TargetPoint.Y = 0;
-	LogoTextMove.CurrentPoint.X = 129;
-	LogoTextMove.CurrentPoint.Y = 0;
-	
-	WelcomeTextMove.TargetPoint.X = 128;
-	WelcomeTextMove.TargetPoint.Y = 0;
-	WelcomeTextMove.CurrentPoint.X = 128;
-	WelcomeTextMove.CurrentPoint.Y = 0;
-}
@@ -607,3 +589,0 @@ void MainAuxFunc(void){
-#if 0
-//主菜单的菜单项
-MenuItem MainMenuItems[] = {
@@ -611,7 +590,0 @@ MenuItem MainMenuItems[] = {
-	{.General_item_text = "Settings",.General_callback = NULL,.General_SubMenuPage = &SettingsMenuPage,.Tiles_Icon = Image_setings},
-	{.General_item_text = "WeChat",.General_callback = NULL,.General_SubMenuPage = NULL,.Tiles_Icon = Image_wechat},
-	{.General_item_text = "Alipay",.General_callback = NULL,.General_SubMenuPage = NULL,.Tiles_Icon = Image_alipay},
-	{.General_item_text = "计算器 Calc 长文本测试 LongText",.General_callback = NULL,.General_SubMenuPage = NULL,.Tiles_Icon = Image_calc},
-	{.General_item_text = "Night",.General_callback = NULL,.General_SubMenuPage = NULL,.Tiles_Icon = Image_night},
-	{.General_item_text = "More",.General_callback = NULL,.General_SubMenuPage = &MoreMenuPage,.Tiles_Icon = Image_more},
-	{.General_item_text = NULL},/*最后一项的General_item_text置为NULL，表示该项为分割线*/
@@ -619,10 +591,0 @@ MenuItem MainMenuItems[] = {
-};
-//设置菜单项内容数组
-MenuItem SettingsMenuItems[] = {
-	{.General_item_text = "亮度",.General_callback = BrightnessWindow,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "黑暗模式",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = &ColorMode},
-	{.General_item_text = "显示帧率",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = &OLED_UI_ShowFps},
-	{.General_item_text = "此设备",.General_callback = NULL,.General_SubMenuPage = &AboutThisDeviceMenuPage,.List_BoolRadioBox = NULL},
-	{.General_item_text = "关于OLED UI",.General_callback = NULL,.General_SubMenuPage = &AboutOLED_UIMenuPage,.List_BoolRadioBox = NULL},
-	{.General_item_text = "感谢观看,一键三连! Thanks for watching, three clicks!",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "[返回]",.General_callback = OLED_UI_Back,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
@@ -630,2 +592,0 @@ MenuItem SettingsMenuItems[] = {
-	{.General_item_text = NULL},/*最后一项的General_item_text置为NULL，表示该项为分割线*/
-};
@@ -633,10 +593,0 @@ MenuItem SettingsMenuItems[] = {
-MenuItem AboutThisDeviceMenuItems[] = {
-	{.General_item_text = "-[MCU:]",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = " STM32F103",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = " RAM:20KB",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = " ROM:64KB",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "-[Screen:]",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = " SSD1306 128x64 OLED",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "-[CP:]",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = " SoftWare SPI",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "[返回]",.General_callback = OLED_UI_Back,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
@@ -644,2 +594,0 @@ MenuItem AboutThisDeviceMenuItems[] = {
-	{.General_item_text = NULL},/*最后一项的General_item_text置为NULL，表示该项为分割线*/
-};
@@ -647,5 +595,0 @@ MenuItem AboutThisDeviceMenuItems[] = {
-MenuItem AboutOLED_UIMenuItems[] = {
-	{.General_item_text = "-[Author:]",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = " bilibili @上nm网课呢",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "-[Adress:]",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = " https://github.com/bdth-7777777/OLED_UI",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
@@ -653 +596,0 @@ MenuItem AboutOLED_UIMenuItems[] = {
-	{.General_item_text = "[返回]",.General_callback = OLED_UI_Back,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
@@ -655,2 +597,0 @@ MenuItem AboutOLED_UIMenuItems[] = {
-	{.General_item_text = NULL},/*最后一项的General_item_text置为NULL，表示该项为分割线*/
-};
@@ -658,32 +598,0 @@ MenuItem AboutOLED_UIMenuItems[] = {
-MenuItem MoreMenuItems[] = {
-	{.General_item_text = "[返回]",.General_callback = OLED_UI_Back,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "字体高度8demo",.General_callback = NULL,.General_SubMenuPage = &Font8MenuPage,.List_BoolRadioBox = NULL},
-	{.General_item_text = "字体高度12demo",.General_callback = NULL,.General_SubMenuPage = &Font12MenuPage,.List_BoolRadioBox = NULL},
-	{.General_item_text = "字体高度16demo",.General_callback = NULL,.General_SubMenuPage = &Font16MenuPage,.List_BoolRadioBox = NULL},
-	{.General_item_text = "字体高度20demo",.General_callback = NULL,.General_SubMenuPage = &Font20MenuPage,.List_BoolRadioBox = NULL},
-	{.General_item_text = "超长文本demo",.General_callback = NULL,.General_SubMenuPage = &LongMenuPage,.List_BoolRadioBox = NULL},
-	{.General_item_text = "回弹动画demo",.General_callback = NULL,.General_SubMenuPage = &SpringMenuPage,.List_BoolRadioBox = NULL},
-	{.General_item_text = "超长菜单列表demo",.General_callback = NULL,.General_SubMenuPage = &LongListMenuPage,.List_BoolRadioBox = NULL},
-	{.General_item_text = "超小区域菜单demo",.General_callback = NULL,.General_SubMenuPage = &SmallAreaMenuPage,.List_BoolRadioBox = NULL},
-	{.General_item_text = "空窗口demo",.General_callback = EmptyWindow,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "文字窗口demo",.General_callback = ShowTextWindow,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "浮点数进度条窗口demo",.General_callback = ShowFloatDataWindow,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "整数进度条窗口demo",.General_callback = ShowIntDataWindow,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = NULL},/*最后一项的General_item_text置为NULL，表示该项为分割线*/
-};
-MenuItem Font8MenuItems[] = {
-	{.General_item_text = "[返回]",.General_callback = OLED_UI_Back,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "中文文本",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "English Text",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "1234567890",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "ABCDEFGHIJKLMNOPQRSTUVWXYZ",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "abcdefghijklmnopqrstuvwxyz",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = ",.[]!@#$+-/^&*()",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "中文文本",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "English Text",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "1234567890",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "ABCDEFGHIJKLMNOPQRSTUVWXYZ",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "abcdefghijklmnopqrstuvwxyz",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = ",.[]!@#$+-/^&*()",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = NULL},/*最后一项的General_item_text置为NULL，表示该项为分割线*/
-};
@@ -691,16 +599,0 @@ MenuItem Font8MenuItems[] = {
-MenuItem Font12MenuItems[] = {
-	{.General_item_text = "[返回]",.General_callback = OLED_UI_Back,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "中文文本",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "English Text",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "1234567890",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "ABCDEFGHIJKLMNOPQRSTUVWXYZ",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "abcdefghijklmnopqrstuvwxyz",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = ",.[]!@#$+-/^&*()",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "中文文本",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "English Text",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "1234567890",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "ABCDEFGHIJKLMNOPQRSTUVWXYZ",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "abcdefghijklmnopqrstuvwxyz",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = ",.[]!@#$+-/^&*()",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = NULL},/*最后一项的General_item_text置为NULL，表示该项为分割线*/
-};
@@ -708,16 +600,0 @@ MenuItem Font12MenuItems[] = {
-MenuItem Font16MenuItems[] = {
-	{.General_item_text = "[返回]",.General_callback = OLED_UI_Back,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "中文文本",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "English Text",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "1234567890",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "ABCDEFGHIJKLMNOPQRSTUVWXYZ",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "abcdefghijklmnopqrstuvwxyz",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = ",.[]!@#$+-/^&*()",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "中文文本",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "English Text",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "1234567890",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "ABCDEFGHIJKLMNOPQRSTUVWXYZ",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "abcdefghijklmnopqrstuvwxyz",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = ",.[]!@#$+-/^&*()",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = NULL},/*最后一项的General_item_text置为NULL，表示该项为分割线*/
-};
@@ -725,16 +601,0 @@ MenuItem Font16MenuItems[] = {
-MenuItem Font20MenuItems[] = {
-	{.General_item_text = "[返回]",.General_callback = OLED_UI_Back,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "中文文本",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "English Text",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "1234567890",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "ABCDEFGHIJKLMNOPQRSTUVWXYZ",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "abcdefghijklmnopqrstuvwxyz",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = ",.[]!@#$+-/^&*()",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "中文文本",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "English Text",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "1234567890",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "ABCDEFGHIJKLMNOPQRSTUVWXYZ",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "abcdefghijklmnopqrstuvwxyz",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = ",.[]!@#$+-/^&*()",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = NULL},/*最后一项的General_item_text置为NULL，表示该项为分割线*/
-};
@@ -742,8 +602,0 @@ MenuItem Font20MenuItems[] = {
-MenuItem LongMenuItems[] = {
-	{.General_item_text = "[返回]",.General_callback = OLED_UI_Back,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "非常非常非常长中文文本",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Very Very Very Long English Text",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "1 2 3 4 5 6 7 8 9 10 11 12 13 14 15",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = NULL},/*最后一项的General_item_text置为NULL，表示该项为分割线*/
-};
@@ -751,16 +603,0 @@ MenuItem LongMenuItems[] = {
-MenuItem SpringMenuItems[] = {
-	{.General_item_text = "[返回]",.General_callback = OLED_UI_Back,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "中文文本",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "English Text",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "1234567890",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "ABCDEFGHIJKLMNOPQRSTUVWXYZ",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "abcdefghijklmnopqrstuvwxyz",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = ",.[]!@#$+-/^&*()",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "中文文本",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "English Text",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "1234567890",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "ABCDEFGHIJKLMNOPQRSTUVWXYZ",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "abcdefghijklmnopqrstuvwxyz",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = ",.[]!@#$+-/^&*()",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = NULL},/*最后一项的General_item_text置为NULL，表示该项为分割线*/
-};
@@ -768,134 +604,0 @@ MenuItem SpringMenuItems[] = {
-MenuItem LongListMenuItems[] = {
-	{.General_item_text = "[返回]",.General_callback = OLED_UI_Back,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item1",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item2",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item3",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item4",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item5",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item6",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item7",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item8",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item9",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item10",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item11",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item12",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item13",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item14",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item15",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item16",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item17",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item18",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item19",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item20",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item21",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item22",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item23",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item24",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item25",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item26",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item27",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item28",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item29",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item30",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item31",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item32",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item33",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item34",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item35",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item36",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item37",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item38",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item39",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item40",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item41",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item42",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item43",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item44",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item45",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item46",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item47",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item48",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item49",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item50",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item51",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item52",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item53",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item54",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item55",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item56",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item57",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item58",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item59",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item60",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item61",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item62",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item63",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item64",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item65",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item66",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item67",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item68",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item69",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item70",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item71",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item72",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item73",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item74",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item75",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item76",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item77",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item78",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item79",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item80",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item81",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item82",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item83",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item84",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item85",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},	
-	{.General_item_text = "Item86",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item87",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item88",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item89",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item90",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item91",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item92",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item93",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item94",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item95",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item96",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item97",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item98",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item99",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item100",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item101",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item102",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item103",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item104",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item105",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item106",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item107",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item108",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item109",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item110",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item111",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item112",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item113",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item114",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item115",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item116",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item117",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item118",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item119",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item120",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item121",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item122",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item123",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item124",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item125",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item126",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item127",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item128",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "Item129",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	
-	{.General_item_text = NULL},/*最后一项的General_item_text置为NULL，表示该项为分割线*/
-};
@@ -903,17 +606 @@ MenuItem LongListMenuItems[] = {
-MenuItem SmallAreaMenuItems[] = {
-	{.General_item_text = "[返回]",.General_callback = OLED_UI_Back,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "中文文本",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "English Text",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "1234567890",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "ABCDEFGHIJKLMNOPQRSTUVWXYZ",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "abcdefghijklmnopqrstuvwxyz",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = ",.[]!@#$+-/^&*()",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "中文文本",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "English Text",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "1234567890",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "ABCDEFGHIJKLMNOPQRSTUVWXYZ",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "abcdefghijklmnopqrstuvwxyz",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = ",.[]!@#$+-/^&*()",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = NULL},/*最后一项的General_item_text置为NULL，表示该项为分割线*/
-};
-#endif
+
@@ -923,31 +610,39 @@ MenuItem MainMenuItems[] = {
-	{.General_item_text = "PID Param",.General_callback = NULL,.General_SubMenuPage = &PIDParamMenuPage,.Tiles_Icon = Image_pid},
-	{.General_item_text = "Other Param",.General_callback = NULL,.General_SubMenuPage = &OtherParamMenuPage,.Tiles_Icon = Image_other},
-	{.General_item_text = "Exit",.General_callback = NULL,.General_SubMenuPage = &ExitMenuPage,.Tiles_Icon = Image_next},
-	{.General_item_text = NULL},/*最后一项的General_item_text置为NULL，表示该项为分割线*/
-};
-//PID参数的菜单项
-MenuItem PIDParamMenuItems[] = {
-	{.General_item_text = "line_kp",.General_callback = Show_line_kp_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "line_ki",.General_callback = Show_line_ki_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "line_kd",.General_callback = Show_line_kd_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "angle_kp",.General_callback = Show_angle_kp_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "angle_ki",.General_callback = Show_angle_ki_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "angle_kd",.General_callback = Show_angle_kd_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "speed_left_kp",.General_callback = Show_speed_left_kp_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "speed_left_ki",.General_callback = Show_speed_left_ki_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "speed_left_kd",.General_callback = Show_speed_left_kd_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "speed_right_kp",.General_callback = Show_speed_right_kp_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
- {.General_item_text = "speed_right_ki",.General_callback = Show_speed_right_ki_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
- {.General_item_text = "speed_right_kd",.General_callback = Show_speed_right_kd_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
- {.General_item_text = NULL},/*最后一项的General_item_text置为NULL，表示该项为分割线*/
-};
-//其他参数的菜单项
-MenuItem OtherParamMenuItems[] = {
-	{.General_item_text = "angle_offset",.General_callback = Show_angle_offset_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "encoder_cnt_odd",.General_callback = Show_encoder_cnt_odd_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = "encoder_cnt_even",.General_callback = Show_encoder_cnt_even_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
-	{.General_item_text = NULL},/*最后一项的General_item_text置为NULL，表示该项为分割线*/
-};
-//Next的菜单项
-MenuItem ExitMenuItems[] = {
-	{.General_item_text = "exit&save",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = &oled_ui_exit},
+	{.General_item_text = "R_ChuFa_x",.General_callback = Show_R_ChuFa_x_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "R_ChuFa_y",.General_callback = Show_R_ChuFa_y_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "B_ChuFa_x",.General_callback = Show_B_ChuFa_x_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "B_ChuFa_y",.General_callback = Show_B_ChuFa_y_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "-------------",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+
+	{.General_item_text = "R_YuanPan_y",.General_callback = Show_R_YuanPan_y_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "B_YuanPan_y",.General_callback = Show_B_YuanPan_y_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "-------------",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+
+	{.General_item_text = "C_GetTarget_y",.General_callback = Show_C_GetTarget_y_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "C_GetTarget_x",.General_callback = Show_C_GetTarget_x_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "-------------",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+
+	{.General_item_text = "C_JieTi_Left_x",.General_callback = Show_C_JieTi_Left_x_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "C_Jieti_FwdTarget_y",.General_callback = Show_C_Jieti_FwdTarget_y_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "C_JitTi_Step_x",.General_callback = Show_C_JitTi_Step_x_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "C_Jieti_Cross_x",.General_callback = Show_C_Jieti_Cross_x_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "C_JieTi_JiaoZhun_x1",.General_callback = Show_C_JieTi_JiaoZhun_x1_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "C_JieTi_JiaoZhun_x2",.General_callback = Show_C_JieTi_JiaoZhun_x2_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "C_JieTi_JiaoZhun_y1",.General_callback = Show_C_JieTi_JiaoZhun_y1_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "C_JieTi_JiaoZhun_y2",.General_callback = Show_C_JieTi_JiaoZhun_y2_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "-------------",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	
+	{.General_item_text = "R_LiZhuang_Delay_x",.General_callback = Show_R_LiZhuang_Delay_x_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "B_LiZhuang_Delay_x",.General_callback = Show_B_LiZhuang_Delay_x_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "C_LiZhuang_Target_y",.General_callback = Show_C_LiZhuang_Target_y_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "C_LiZhuang_Fwd_y",.General_callback = Show_C_LiZhuang_Fwd_y_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "R_LiZhuang_Back_y",.General_callback = Show_R_LiZhuang_Back_y_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "B_LiZhuang_Back_y",.General_callback = Show_B_LiZhuang_Back_y_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "-------------",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	
+	{.General_item_text = "R_HuiJia_x",.General_callback = Show_R_HuiJia_x_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "B_HuiJia_x",.General_callback = Show_B_HuiJia_x_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "C_HuiJia_y",.General_callback = Show_C_HuiJia_y_Window,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	{.General_item_text = "-------------",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = NULL},
+	
+	{.General_item_text = "exit&save",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = &oled_ui_exit_save},
+	{.General_item_text = "exit&cancel",.General_callback = NULL,.General_SubMenuPage = NULL,.List_BoolRadioBox = &oled_ui_exit_cancel},
@@ -960,61 +654,0 @@ MenuItem ExitMenuItems[] = {
-#if 0
-MenuPage MainMenuPage = {
-	//通用属性，必填
-	.General_MenuType = MENU_TYPE_TILES,  		 //菜单类型为磁贴类型
-	.General_CursorStyle = NOT_SHOW,			 //光标类型
-	.General_FontSize = OLED_UI_FONT_16,			//字高
-	.General_ParentMenuPage = NULL,				//由于这是根菜单，所以父菜单为NULL
-	.General_LineSpace = 5,						//磁贴间距 单位：像素（对于磁贴类型菜单，此值表示每个磁贴之间的间距，对于列表类型菜单，此值表示行间距）
-	.General_MoveStyle = UNLINEAR,				//移动方式
-	.General_MovingSpeed = SPEED,					//动画移动速度(此值根据实际效果调整)
-	.General_ShowAuxiliaryFunction = MainAuxFunc,		 //显示辅助函数
-	.General_MenuItems = MainMenuItems,			//菜单项内容数组
-
-	//特殊属性，根据.General_MenuType的类型选择
-	.Tiles_ScreenHeight = 64,					//屏幕高度
-	.Tiles_ScreenWidth = 128,						//屏幕宽度
-	.Tiles_TileWidth = 32,						 //磁贴宽度
-	.Tiles_TileHeight = 32,						 //磁贴高度
-};
-
-
-MenuPage SettingsMenuPage = {
-	//通用属性，必填
-	.General_MenuType = MENU_TYPE_LIST,  		 //菜单类型为列表类型
-	.General_CursorStyle = REVERSE_ROUNDRECTANGLE,	 //光标类型为线型
-	.General_FontSize = OLED_UI_FONT_12,			//字高
-	.General_ParentMenuPage = &MainMenuPage,		 //父菜单为主菜单
-	.General_LineSpace = 4,						//行间距 单位：像素
-	.General_MoveStyle = UNLINEAR,				//移动方式为非线性曲线动画
-	.General_MovingSpeed = SPEED,					//动画移动速度(此值根据实际效果调整)
-	.General_ShowAuxiliaryFunction = SettingAuxFunc,		 //显示辅助函数
-	.General_MenuItems = SettingsMenuItems,		 //菜单项内容数组
-
-	//特殊属性，根据.General_MenuType的类型选择
-	.List_MenuArea = {32, 0, 95, 64},			 //列表显示区域
-	.List_IfDrawFrame = false,					 //是否显示边框
-	.List_IfDrawLinePerfix = true,				 //是否显示行前缀
-	.List_StartPointX = 4,                        //列表起始点X坐标
-	.List_StartPointY = 2,                        //列表起始点Y坐标
-};
-
-MenuPage AboutThisDeviceMenuPage = {
-	//通用属性，必填
-	.General_MenuType = MENU_TYPE_LIST,  		 //菜单类型为列表类型
-	.General_CursorStyle = REVERSE_BLOCK,	 //光标类型为圆角矩形
-	.General_FontSize = OLED_UI_FONT_12,			//字高
-	.General_ParentMenuPage = &SettingsMenuPage,		 //父菜单为主菜单
-	.General_LineSpace = 4,						//行间距 单位：像素
-	.General_MoveStyle = UNLINEAR,				//移动方式为非线性曲线动画
-	.General_MovingSpeed = SPEED,					//动画移动速度(此值根据实际效果调整)
-	.General_ShowAuxiliaryFunction = AboutThisDeviceAuxFunc,		 //显示辅助函数
-	.General_MenuItems = AboutThisDeviceMenuItems,		 //菜单项内容数组
-
-	//特殊属性，根据.General_MenuType的类型选择
-	.List_MenuArea = {0, 0, 100, 64},			 //列表显示区域
-	.List_IfDrawFrame = false,					 //是否显示边框
-	.List_IfDrawLinePerfix = false,				 //是否显示行前缀
-	.List_StartPointX = 4,                        //列表起始点X坐标
-	.List_StartPointY = 2,                        //列表起始点Y坐标
-
-};
@@ -1022,209 +655,0 @@ MenuPage AboutThisDeviceMenuPage = {
-MenuPage AboutOLED_UIMenuPage = {
-	//通用属性，必填
-	.General_MenuType = MENU_TYPE_LIST,  		 //菜单类型为列表类型
-	.General_CursorStyle = REVERSE_BLOCK,	 //光标类型为圆角矩形
-	.General_FontSize = OLED_UI_FONT_12,			//字高
-	.General_ParentMenuPage = &SettingsMenuPage,		 //父菜单为主菜单
-	.General_LineSpace = 4,						//行间距 单位：像素
-	.General_MoveStyle = UNLINEAR,				//移动方式为非线性曲线动画
-	.General_MovingSpeed = SPEED,					//动画移动速度(此值根据实际效果调整)
-	.General_ShowAuxiliaryFunction = AboutOLED_UIAuxFunc,		 //显示辅助函数
-	.General_MenuItems = AboutOLED_UIMenuItems,		 //菜单项内容数组
-
-	//特殊属性，根据.General_MenuType的类型选择
-	.List_MenuArea = {0, 0, 105, 64},			 //列表显示区域
-	.List_IfDrawFrame = false,					 //是否显示边框
-	.List_IfDrawLinePerfix = false,				 //是否显示行前缀
-	.List_StartPointX = 4,                        //列表起始点X坐标
-	.List_StartPointY = 2,                        //列表起始点Y坐标
-
-};
-MenuPage MoreMenuPage = {
-	//通用属性，必填
-	.General_MenuType = MENU_TYPE_LIST,  		 //菜单类型为列表类型
-	.General_CursorStyle = REVERSE_ROUNDRECTANGLE,	 //光标类型为圆角矩形
-	.General_FontSize = OLED_UI_FONT_12,			//字高
-	.General_ParentMenuPage = &MainMenuPage,		 //父菜单为主菜单
-	.General_LineSpace = 4,						//行间距 单位：像素
-	.General_MoveStyle = UNLINEAR,				//移动方式为非线性曲线动画
-	.General_MovingSpeed = SPEED,					//动画移动速度(此值根据实际效果调整)
-	.General_ShowAuxiliaryFunction = NULL,		 //显示辅助函数
-	.General_MenuItems = MoreMenuItems,		 //菜单项内容数组
-
-	//特殊属性，根据.General_MenuType的类型选择
-	.List_MenuArea = {1, 1, 128, 64},			 //列表显示区域
-	.List_IfDrawFrame = false,					 //是否显示边框
-	.List_IfDrawLinePerfix = true,				 //是否显示行前缀
-	.List_StartPointX = 4,                        //列表起始点X坐标
-	.List_StartPointY = 2,                        //列表起始点Y坐标
-
-};
-
-MenuPage Font8MenuPage = {
-	//通用属性，必填
-	.General_MenuType = MENU_TYPE_LIST,  		 //菜单类型为列表类型
-	.General_CursorStyle = REVERSE_ROUNDRECTANGLE,	 //光标类型为圆角矩形
-	.General_FontSize = OLED_UI_FONT_8,			//字高
-	.General_ParentMenuPage = &MoreMenuPage,		 //父菜单为主菜单
-	.General_LineSpace = 5,						//行间距 单位：像素
-	.General_MoveStyle = UNLINEAR,				//移动方式为非线性曲线动画
-	.General_MovingSpeed = SPEED,					//动画移动速度(此值根据实际效果调整)
-	.General_ShowAuxiliaryFunction = NULL,		 //显示辅助函数
-	.General_MenuItems = Font8MenuItems,		 //菜单项内容数组
-
-	//特殊属性，根据.General_MenuType的类型选择
-	.List_MenuArea = {0, 0, 128, 64},			 //列表显示区域
-	.List_IfDrawFrame = false,					 //是否显示边框
-	.List_IfDrawLinePerfix = true,				 //是否显示行前缀
-	.List_StartPointX = 4,                        //列表起始点X坐标
-	.List_StartPointY = 2,                        //列表起始点Y坐标
-
-};
-
-MenuPage Font12MenuPage = {
-	//通用属性，必填
-	.General_MenuType = MENU_TYPE_LIST,  		 //菜单类型为列表类型
-	.General_CursorStyle = REVERSE_ROUNDRECTANGLE,	 //光标类型为圆角矩形
-	.General_FontSize = OLED_UI_FONT_12,			//字高
-	.General_ParentMenuPage = &MoreMenuPage,		 //父菜单为主菜单
-	.General_LineSpace = 4,						//行间距 单位：像素
-	.General_MoveStyle = UNLINEAR,				//移动方式为非线性曲线动画
-	.General_MovingSpeed = SPEED,					//动画移动速度(此值根据实际效果调整)
-	.General_ShowAuxiliaryFunction = NULL,		 //显示辅助函数
-	.General_MenuItems = Font12MenuItems,		 //菜单项内容数组
-
-	//特殊属性，根据.General_MenuType的类型选择
-	.List_MenuArea = {0, 0, 128, 64},			 //列表显示区域
-	.List_IfDrawFrame = false,					 //是否显示边框
-	.List_IfDrawLinePerfix = true,				 //是否显示行前缀
-	.List_StartPointX = 4,                        //列表起始点X坐标
-	.List_StartPointY = 2,                        //列表起始点Y坐标
-
-};
-
-MenuPage Font16MenuPage = {
-	//通用属性，必填
-	.General_MenuType = MENU_TYPE_LIST,  		 //菜单类型为列表类型
-	.General_CursorStyle = REVERSE_ROUNDRECTANGLE,	 //光标类型为圆角矩形
-	.General_FontSize = OLED_UI_FONT_16,			//字高
-	.General_ParentMenuPage = &MoreMenuPage,		 //父菜单为主菜单
-	.General_LineSpace = 5,						//行间距 单位：像素
-	.General_MoveStyle = UNLINEAR,				//移动方式为非线性曲线动画
-	.General_MovingSpeed = SPEED,					//动画移动速度(此值根据实际效果调整)
-	.General_ShowAuxiliaryFunction = NULL,		 //显示辅助函数
-	.General_MenuItems = Font16MenuItems,		 //菜单项内容数组
-
-	//特殊属性，根据.General_MenuType的类型选择
-	.List_MenuArea = {1, 1, 126, 62},			 //列表显示区域
-	.List_IfDrawFrame = false,					 //是否显示边框
-	.List_IfDrawLinePerfix = true,				 //是否显示行前缀
-	.List_StartPointX = 4,                        //列表起始点X坐标
-	.List_StartPointY = 2,                        //列表起始点Y坐标
-
-};
-
-MenuPage Font20MenuPage = {
-	//通用属性，必填
-	.General_MenuType = MENU_TYPE_LIST,  		 //菜单类型为列表类型
-	.General_CursorStyle = REVERSE_ROUNDRECTANGLE,	 //光标类型为圆角矩形
-	.General_FontSize = OLED_UI_FONT_20,			//字高
-	.General_ParentMenuPage = &MoreMenuPage,		 //父菜单为主菜单
-	.General_LineSpace = 10,						//行间距 单位：像素
-	.General_MoveStyle = UNLINEAR,				//移动方式为非线性曲线动画
-	.General_MovingSpeed = SPEED,					//动画移动速度(此值根据实际效果调整)
-	.General_ShowAuxiliaryFunction = NULL,		 //显示辅助函数
-	.General_MenuItems = Font20MenuItems,		 //菜单项内容数组
-
-	//特殊属性，根据.General_MenuType的类型选择
-	.List_MenuArea = {0, 0, 128, 64},			 //列表显示区域
-	.List_IfDrawFrame = false,					 //是否显示边框
-	.List_IfDrawLinePerfix = true,				 //是否显示行前缀
-	.List_StartPointX = 4,                        //列表起始点X坐标
-	.List_StartPointY = 2,                        //列表起始点Y坐标
-
-};
-
-MenuPage LongMenuPage = {
-	//通用属性，必填
-	.General_MenuType = MENU_TYPE_LIST,  		 //菜单类型为列表类型
-	.General_CursorStyle = REVERSE_ROUNDRECTANGLE,	 //光标类型为圆角矩形
-	.General_FontSize = OLED_UI_FONT_12,			//字高
-	.General_ParentMenuPage = &MoreMenuPage,		 //父菜单为主菜单
-	.General_LineSpace = 4,						//行间距 单位：像素
-	.General_MoveStyle = UNLINEAR,				//移动方式为非线性曲线动画
-	.General_MovingSpeed = SPEED,					//动画移动速度(此值根据实际效果调整)
-	.General_ShowAuxiliaryFunction = NULL,		 //显示辅助函数
-	.General_MenuItems = LongMenuItems,		 //菜单项内容数组
-
-	//特殊属性，根据.General_MenuType的类型选择
-	.List_MenuArea = {0, 0, 128, 64},			 //列表显示区域
-	.List_IfDrawFrame = false,					 //是否显示边框
-	.List_IfDrawLinePerfix = true,				 //是否显示行前缀
-	.List_StartPointX = 4,                        //列表起始点X坐标
-	.List_StartPointY = 2,                        //列表起始点Y坐标
-
-};
-
-MenuPage SpringMenuPage = {
-	//通用属性，必填
-	.General_MenuType = MENU_TYPE_LIST,  		 //菜单类型为列表类型
-	.General_CursorStyle = REVERSE_ROUNDRECTANGLE,	 //光标类型为圆角矩形
-	.General_FontSize = OLED_UI_FONT_12,			//字高
-	.General_ParentMenuPage = &MoreMenuPage,		 //父菜单为主菜单
-	.General_LineSpace = 4,						//行间距 单位：像素
-	.General_MoveStyle = PID_CURVE,				//移动方式为非线性曲线动画
-	.General_MovingSpeed = SPEED,					//动画移动速度(此值根据实际效果调整)
-	.General_ShowAuxiliaryFunction = NULL,		 //显示辅助函数
-	.General_MenuItems = SpringMenuItems,		 //菜单项内容数组
-
-	//特殊属性，根据.General_MenuType的类型选择
-	.List_MenuArea = {0, 0, 128, 64},			 //列表显示区域
-	.List_IfDrawFrame = false,					 //是否显示边框
-	.List_IfDrawLinePerfix = true,				 //是否显示行前缀
-	.List_StartPointX = 4,                        //列表起始点X坐标
-	.List_StartPointY = 2,                        //列表起始点Y坐标
-
-};
-
-MenuPage LongListMenuPage = {
-	//通用属性，必填
-	.General_MenuType = MENU_TYPE_LIST,  		 //菜单类型为列表类型
-	.General_CursorStyle = REVERSE_ROUNDRECTANGLE,	 //光标类型为圆角矩形
-	.General_FontSize = OLED_UI_FONT_12,			//字高
-	.General_ParentMenuPage = &MoreMenuPage,		 //父菜单为主菜单
-	.General_LineSpace = 4,						//行间距 单位：像素
-	.General_MoveStyle = UNLINEAR,				//移动方式为非线性曲线动画
-	.General_MovingSpeed = SPEED,					//动画移动速度(此值根据实际效果调整)
-	.General_ShowAuxiliaryFunction = NULL,		 //显示辅助函数
-	.General_MenuItems = LongListMenuItems,		 //菜单项内容数组
-
-	//特殊属性，根据.General_MenuType的类型选择
-	.List_MenuArea = {0, 0, 128, 64},			 //列表显示区域
-	.List_IfDrawFrame = false,					 //是否显示边框
-	.List_IfDrawLinePerfix = true,				 //是否显示行前缀
-	.List_StartPointX = 4,                        //列表起始点X坐标
-	.List_StartPointY = 2,                        //列表起始点Y坐标
-
-};
-
-MenuPage SmallAreaMenuPage = {
-	//通用属性，必填
-	.General_MenuType = MENU_TYPE_LIST,  		 //菜单类型为列表类型
-	.General_CursorStyle = REVERSE_ROUNDRECTANGLE,	 //光标类型为圆角矩形
-	.General_FontSize = OLED_UI_FONT_12,			//字高
-	.General_ParentMenuPage = &MoreMenuPage,		 //父菜单为主菜单
-	.General_LineSpace = 6,						//行间距 单位：像素
-	.General_MoveStyle = UNLINEAR,				//移动方式为非线性曲线动画
-	.General_MovingSpeed = SPEED,					//动画移动速度(此值根据实际效果调整)
-	.General_ShowAuxiliaryFunction = NULL,		 //显示辅助函数
-	.General_MenuItems = SmallAreaMenuItems,		 //菜单项内容数组
-
-	//特殊属性，根据.General_MenuType的类型选择
-	.List_MenuArea = {10, 10, 60, 36},			 //列表显示区域
-	.List_IfDrawFrame = true,					 //是否显示边框
-	.List_IfDrawLinePerfix = true,				 //是否显示行前缀
-	.List_StartPointX = 4,                        //列表起始点X坐标
-	.List_StartPointY = 2,                        //列表起始点Y坐标
-
-};
-#endif
@@ -1232,18 +656,0 @@ MenuPage SmallAreaMenuPage = {
-MenuPage MainMenuPage = {
-	//通用属性，必填
-	.General_MenuType = MENU_TYPE_TILES,  		 //菜单类型为磁贴类型
-	.General_CursorStyle = NOT_SHOW,			 //光标类型
-	.General_FontSize = OLED_UI_FONT_16,			//字高
-	.General_ParentMenuPage = NULL,				//由于这是根菜单，所以父菜单为NULL
-	.General_LineSpace = 5,						//磁贴间距 单位：像素（对于磁贴类型菜单，此值表示每个磁贴之间的间距，对于列表类型菜单，此值表示行间距）
-	.General_MoveStyle = UNLINEAR,				//移动方式
-	.General_MovingSpeed = SPEED,					//动画移动速度(此值根据实际效果调整)
-	.General_ShowAuxiliaryFunction = MainAuxFunc,		 //显示辅助函数
-	.General_MenuItems = MainMenuItems,			//菜单项内容数组
-
-	//特殊属性，根据.General_MenuType的类型选择
-	.Tiles_ScreenHeight = 64,					//屏幕高度
-	.Tiles_ScreenWidth = 128,						//屏幕宽度
-	.Tiles_TileWidth = 32,						 //磁贴宽度
-	.Tiles_TileHeight = 32,						 //磁贴高度
-};
@@ -1251,21 +658 @@ MenuPage MainMenuPage = {
-MenuPage PIDParamMenuPage = {
-	//通用属性，必填
-	.General_MenuType = MENU_TYPE_LIST,  		 //菜单类型为列表类型
-	.General_CursorStyle = REVERSE_ROUNDRECTANGLE,	 //光标类型为圆角矩形
-	.General_FontSize = OLED_UI_FONT_12,			//字高
-	.General_ParentMenuPage = &MainMenuPage,		 //父菜单为主菜单
-	.General_LineSpace = 6,						//行间距 单位：像素
-	.General_MoveStyle = UNLINEAR,				//移动方式为非线性曲线动画
-	.General_MovingSpeed = SPEED,					//动画移动速度(此值根据实际效果调整)
-	.General_ShowAuxiliaryFunction = NULL,		 //显示辅助函数
-	.General_MenuItems = PIDParamMenuItems,		 //菜单项内容数组
-
-	//特殊属性，根据.General_MenuType的类型选择
-	.List_MenuArea = {0, 0, 128, 64},			 //列表显示区域
-	.List_IfDrawFrame = true,					 //是否显示边框
-	.List_IfDrawLinePerfix = true,				 //是否显示行前缀
-	.List_StartPointX = 4,                        //列表起始点X坐标
-	.List_StartPointY = 2,                        //列表起始点Y坐标
-};
-
-MenuPage OtherParamMenuPage = {
+MenuPage MainMenuPage = {
@@ -1276 +663 @@ MenuPage OtherParamMenuPage = {
-	.General_ParentMenuPage = &MainMenuPage,		 //父菜单为主菜单
+	.General_ParentMenuPage = NULL,		 //父菜单
@@ -1281 +668 @@ MenuPage OtherParamMenuPage = {
-	.General_MenuItems = OtherParamMenuItems,		 //菜单项内容数组
+	.General_MenuItems = MainMenuItems,		 //菜单项内容数组
@@ -1291,19 +677,0 @@ MenuPage OtherParamMenuPage = {
-MenuPage ExitMenuPage = {
-	//通用属性，必填
-	.General_MenuType = MENU_TYPE_LIST,  		 //菜单类型为列表类型
-	.General_CursorStyle = REVERSE_ROUNDRECTANGLE,	 //光标类型为圆角矩形
-	.General_FontSize = OLED_UI_FONT_12,			//字高
-	.General_ParentMenuPage = &MainMenuPage,		 //父菜单为主菜单
-	.General_LineSpace = 6,						//行间距 单位：像素
-	.General_MoveStyle = UNLINEAR,				//移动方式为非线性曲线动画
-	.General_MovingSpeed = SPEED,					//动画移动速度(此值根据实际效果调整)
-	.General_ShowAuxiliaryFunction = NULL,		 //显示辅助函数
-	.General_MenuItems = ExitMenuItems,		 //菜单项内容数组
-
-	//特殊属性，根据.General_MenuType的类型选择
-	.List_MenuArea = {0, 0, 128, 64},			 //列表显示区域
-	.List_IfDrawFrame = true,					 //是否显示边框
-	.List_IfDrawLinePerfix = true,				 //是否显示行前缀
-	.List_StartPointX = 4,                        //列表起始点X坐标
-	.List_StartPointY = 2,                        //列表起始点Y坐标
-};
